agenda/tests/Feature/UserTest.php

```diff
@@ -0,0 +1,640 @@
+<?php
+
+namespace Tests\Feature;
+
+use Carbon\Carbon;
+use App\Models\ChangePassword;
+
+class UserTest extends BaseTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    /**
+     * Retorna sucesso ao realizar o registro de um novo usuário no sistema
+     */
+    public function testReturnSuccessWhenRegisterNewUser()
+    {
+        $password = $this->faker->password;
+        $body = [
+            'name'             => $this->faker->name,
+            'email'            => $this->faker->email,
+            'password'         => $password,
+            'confirm_password' => $password
+        ];
+
+        $this->postJson(route('users.register'), $body)
+            ->assertHeader('content-type', 'application/json')
+            ->assertStatus(200)
+            ->assertJson([
+                'success' => true,
+                'request' => route('users.register'),
+                'method'  => 'POST',
+                'code'    => 200,
+            ], true);
+    }
+
+    /**
+     * Retorna erro ao tentar registrar um usuário informando senha e senha de
+     * confirmação diferentes
+     */
+    public function testReturnErrorWhenRegisterNewUserWithDifferentPasswords()
+    {
+        $body = [
+            'name'             => $this->faker->name,
+            'email'            => $this->faker->email,
+            'password'         => $this->faker->password,
+            'confirm_password' => $this->faker->password
+        ];
+
+        $this->postJson(route('users.register'), $body)
+            ->assertHeader('content-type', 'application/json')
+            ->assertStatus(422)
+            ->assertJson([
+                'success' => false,
+                'request' => route('users.register'),
+                'method'  => 'POST',
+                'code'    => 422,
+                'data'    => null,
+            ], true)
+            ->assertJsonStructure(['errors'])
+            ->assertJsonFragment([
+                'code' => null
+            ]);
+    }
+
+    /**
+     * Retorna erro ao tentar registrar um usuário com um email já cadastrado
+     */
+    public function testReturnErrorWhenRegisterNewUserWithEqualsEmail()
+    {
+        $generateUser = $this->generateUserAndToken();
```

Comment:
Como a variável já chama generateUser, vc poderia chamar a prop user direito ( $this->generateUserAndToken()->user; ) , o que acha?